plugins/channel_header_plug/channel_header_plug.tsx

```diff
@@ -123,9 +127,22 @@ class ChannelHeaderPlug extends React.PureComponent<ChannelHeaderPlugProps, Chan
         super(props);
         this.state = {
             dropdownOpen: false,
+            disableButtonsClosingRHS: false,
         };
     }
 
+    componentDidUpdate(prevProps: ChannelHeaderPlugProps) {
+        if (prevProps.sidebarOpen && !this.props.sidebarOpen) {
+            // eslint-disable-next-line react/no-did-update-set-state
+            this.setState({disableButtonsClosingRHS: true});
+
+            setTimeout(() => {
+                // eslint-disable-next-line react/no-did-update-set-state
+                this.setState({disableButtonsClosingRHS: false});
```

Comment:
Also something we shouldn't worry about probably, but if the channel_header_plug component gets unmounted before the timer kicks in, we may have a "trying to set state on unmounted object" error.

I don't foresee how the channel_header_plug component may be unmounted, specially in such short notice, but wanted to raise the concern just in case.

Reply:
N/A since this variable has been moved out of the component's state